walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -13,11 +13,210 @@
 // limitations under the License.
 
 package com.google.sps;
+import com.google.sps.*;
+import java.util.*;
+public final class FindMeetingQuery {
 
-import java.util.Collection;
+  /* Given all events, return a HashMap that contains the schedule of mandatory attandees and optional attandees 
+  param: @events: all events containing time and attandees list 
+         @request: Meeting requests containing attandees list
+  return: @allSchedules a hashmap with 2 key: 'mandatory' and 'optional', the values are another HashMap whose key is attandee and 
+          value is the schedule of that attandee
+  */
+  public static HashMap<String, HashMap<String, ArrayList<TimeRange>>> getSchedules(Collection<Event> events, MeetingRequest request) {
+    Collection<String> mandatoryAttandeeSet = request.getAttendees();
+    Collection<String> optionalAttandeeSet = request.getOptionalAttendees();
+    HashMap<String, HashMap<String, ArrayList<TimeRange>>> allSchedules = new HashMap<String, HashMap<String, ArrayList<TimeRange>>>();
+    for (Event event: events) {
+        TimeRange timeRange = event.getWhen();
+        Set<String> allAttandees = event.getAttendees();
+        for (String attandee : allAttandees) {
+            if (mandatoryAttandeeSet.contains(attandee)) {
+                HashMap<String, ArrayList<TimeRange>> mandatorySchedules = allSchedules.getOrDefault("mandatory", new HashMap<String, ArrayList<TimeRange>>());
+                ArrayList<TimeRange> attandeeSchedule = mandatorySchedules.getOrDefault(attandee, new ArrayList<TimeRange>());
+                attandeeSchedule.add(timeRange);
+                mandatorySchedules.put(attandee, attandeeSchedule);
+                allSchedules.put("mandatory", mandatorySchedules);
+            }
+            else if (optionalAttandeeSet.contains(attandee)){
+                HashMap<String, ArrayList<TimeRange>> optionalSchedules = allSchedules.getOrDefault("optional", new HashMap<String, ArrayList<TimeRange>>());
+                ArrayList<TimeRange> attandeeSchedule = optionalSchedules.getOrDefault(attandee, new ArrayList<TimeRange>());
+                attandeeSchedule.add(timeRange);
+                optionalSchedules.put(attandee, attandeeSchedule);
+                allSchedules.put("optional", optionalSchedules);
+            }
+            
+        }
+    }
+    return allSchedules;
+  }
+  /*Given two overlapped TimeRanges, return the merged TimeRange
+  param: @one and @two: overlapped TimeRanges
+  return: @mergeTimeSlot: merge TimeRange
+  */
+  public static TimeRange mergeTwoOverlappedTimeRanges(TimeRange one, TimeRange two) {
```

Comment:
Nit: This function seems to take a union of the two time ranges, but from the name I initially thought it took an intersect of the two ranges. A name like `unionTwoOverlappedTimeRanges` might make the function a bit clearer.